src/pyfmi/fmi3.pyx

```diff
@@ -568,31 +676,78 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
 
         return vr
 
-    cpdef FMIL3.fmi3_base_type_enu_t get_variable_data_type(self, variable_name) except *:
+    cdef FMIL3.fmi3_base_type_enu_t _get_variable_data_type(self, variable_name) except *:
+        cdef FMIL3.fmi3_import_variable_t* variable
+        variable_name = pyfmi_util.encode(variable_name)
+        cdef char* variablename = variable_name
+
+        variable = FMIL3.fmi3_import_get_variable_by_name(self._fmu, variablename)
+        if variable == NULL:
+            raise FMUException(f"The variable {pyfmi_util.decode(variablename)} could not be found.")
+        return FMIL3.fmi3_import_get_variable_base_type(variable)
```

Comment:
Made internal `_get...` and wrapped below, because Cython cannot convert the type `FMIL3.fmi3_base_type_enu_t` directly to `int` otherwise due to the signature of the function.

Reply:
Ah, I see. 

I suppose we can take this separately later. Wondering if it was possible to move the Python enum to the Cython layer.

Reply:
I tried that but it seemed to cause complications and it would simply be easier to have it in pure Python. But if you find any nice way to improve it I'm happy to change it.